exp/controllers/rosamachinepool_controller.go

```diff
@@ -372,6 +379,47 @@ func validateMachinePoolSpec(machinePoolScope *scope.RosaMachinePoolScope) (*str
 	return nil, nil
 }
 
+func (r *ROSAMachinePoolReconciler) reconcileProviderIDList(ctx context.Context, machinePoolScope *scope.RosaMachinePoolScope, nodePool *cmv1.NodePool) error {
+	tags := nodePool.AWSNodePool().Tags()
+	if len(tags) == 0 {
+		// can't identify EC2 instances belonging to this NodePool without tags.
+		return nil
+	}
+
+	ec2Svc := scope.NewEC2Client(machinePoolScope, machinePoolScope, &machinePoolScope.Logger, machinePoolScope.InfraCluster())
+	response, err := ec2Svc.DescribeInstancesWithContext(ctx, &ec2.DescribeInstancesInput{
+		Filters: buildEC2FiltersFromTags(tags),
+	})
+	if err != nil {
+		return err
+	}
+
+	var providerIDList []string
+	for _, reservation := range response.Reservations {
+		for _, instance := range reservation.Instances {
+			providerID := fmt.Sprintf("aws:///%s/%s", *instance.Placement.AvailabilityZone, *instance.InstanceId)
```

Comment:
Is this importable from somewhere? How do we know that this will be the value that ends up in `node.spec.providerID`?

Reply:
this is not importable, but ihttps://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/91941d950a27f94c078e4797c12a856f416b39a7/pkg/cloud/scope/machine.go#L152

Reply:
Could we refactor to make sure our implementation always remains in sync?

Reply:
@vincepri any thoughts/objections on refactoring?

Reply:
To be clear, the "refactor" ask is to export a function that does this formatting ...